File: frontend/src/services/congressionalSession.ts
// Congressional Session Service
// Handles current Congressional session data and context

export interface CongressionalSession {
  sessionId: number;
  congressNumber: number;
  startDate: string;
  endDate: string;
  isCurrent: boolean;
  partyControlHouse: 'Republican' | 'Democratic' | null;
  partyControlSenate: 'Republican' | 'Democratic' | null;
  displayName: string;
  dateRange: string;
}

export interface CongressionalSessionContextProps {
  currentSession: CongressionalSession;
  isUnifiedGovernment: boolean;
  majorityParty: 'Republican' | 'Democratic' | null;
  sessionTransition?: {
    nextCongress: number;
    transitionDate: string;
  };
}

// Current 119th Congress session data
export const CURRENT_CONGRESSIONAL_SESSION: CongressionalSession = {
  sessionId: 119,
  congressNumber: 119,
  startDate: '2025-01-03',
  endDate: '2027-01-03',
  isCurrent: true,
  partyControlHouse: 'Republican',
  partyControlSenate: 'Republican',
  displayName: '119th Congress',
  dateRange: '2025-2027'
};

// Get current Congressional session context
export const getCurrentSessionContext = (): CongressionalSessionContextProps => {
  const session = CURRENT_CONGRESSIONAL_SESSION;
  
  return {
    currentSession: session,
    isUnifiedGovernment: session.partyControlHouse === session.partyControlSenate,
    majorityParty: session.partyControlHouse, // Since both chambers are Republican
    sessionTransition: {
      nextCongress: 120,
      transitionDate: '2027-01-03'
    }
  };
};

// Get Congressional session display string
export const getSessionDisplayString = (): string => {
  const session = CURRENT_CONGRESSIONAL_SESSION;
  return `${session.displayName} (${session.dateRange})`;
};

// Get party control summary
export const getPartyControlSummary = (): string => {
  const session = CURRENT_CONGRESSIONAL_SESSION;
  if (session.partyControlHouse === session.partyControlSenate) {
    return `${session.partyControlHouse} Unified Control`;
  }
  return `${session.partyControlHouse} House, ${session.partyControlSenate} Senate`;
};

// Check if session is current
export const isCurrentSession = (congressNumber: number): boolean => {
  return congressNumber === CURRENT_CONGRESSIONAL_SESSION.congressNumber;
};

// Get leadership context based on party control
export const getLeadershipContext = () => {
  const session = CURRENT_CONGRESSIONAL_SESSION;
  return {
    houseMajority: session.partyControlHouse,
    senateMajority: session.partyControlSenate,
    isUnified: session.partyControlHouse === session.partyControlSenate,
    committeesControlledBy: session.partyControlSenate // Senate committees 
  };
};

const congressionalSessionService = {
  getCurrentSessionContext,
  getSessionDisplayString,
  getPartyControlSummary,
  isCurrentSession,
  getLeadershipContext,
  CURRENT_CONGRESSIONAL_SESSION
};

export default congressionalSessionService;

File: backend/app/schemas/congressional_session.py
"""
Pydantic schemas for congressional sessions.
"""
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, Field


class CongressionalSessionBase(BaseModel):
    """Base schema for congressional sessions."""
    congress_number: int = Field(..., description="Congress number (e.g., 119)")
    start_date: date = Field(..., description="Session start date")
    end_date: date = Field(..., description="Session end date")
    is_current: bool = Field(default=False, description="Whether this is the current session")
    party_control_house: Optional[str] = Field(None, description="Party controlling the House")
    party_control_senate: Optional[str] = Field(None, description="Party controlling the Senate")
    session_name: Optional[str] = Field(None, description="Display name (e.g., '119th Congress')")
    description: Optional[str] = Field(None, description="Session description")
    election_year: Optional[int] = Field(None, description="Election year for this Congress")


class CongressionalSessionCreate(CongressionalSessionBase):
    """Schema for creating congressional sessions."""
    pass


class CongressionalSessionUpdate(BaseModel):
    """Schema for updating congressional sessions."""
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    is_current: Optional[bool] = None
    party_control_house: Optional[str] = None
    party_control_senate: Optional[str] = None
    session_name: Optional[str] = None
    description: Optional[str] = None
    election_year: Optional[int] = None


class CongressionalSession(CongressionalSessionBase):
    """Schema for congressional session responses."""
    session_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    # Computed properties
    display_name: str = Field(..., description="Formatted display name")
    years_display: str = Field(..., description="Years in format '2025-2027'")
    is_republican_controlled_house: bool = Field(..., description="Whether Republicans control House")
    is_republican_controlled_senate: bool = Field(..., description="Whether Republicans control Senate")
    unified_control: bool = Field(..., description="Whether one party controls both chambers")

    class Config:
        from_attributes = True


class CongressionalSessionSummary(BaseModel):
    """Summary schema for congressional sessions."""
    congress_number: int
    display_name: str
    years_display: str
    is_current: bool
    party_control_house: Optional[str] = None
    party_control_senate: Optional[str] = None
    unified_control: bool


class CurrentCongressInfo(BaseModel):
    """Information about the current Congress."""
    current_session: CongressionalSession
    days_remaining: int = Field(..., description="Days remaining in current session")
    next_election_year: int = Field(..., description="Next House election year")
    house_majority: str = Field(..., description="Party with House majority")
    senate_majority: str = Field(..., description="Party with Senate majority")
    unified_government: bool = Field(..., description="Whether same party controls both chambers")
    
    # Historical context
    previous_session: Optional[CongressionalSessionSummary] = None
    next_session: Optional[CongressionalSessionSummary] = None